### PR TITLE
fix: reshape binary validation labels

### LINKED ISSUE #250: Bug: ClassificationTrainer.fit() validation branch doesn't reshape labels for binary classification
### Description
In `ClassificationTrainer.fit()`, the training branch reshapes labels for
binary classification before computing the loss:

```python
if self._num_classes == 2:
    labels = labels.view(-1, 1).float()
```

The validation branch does not do this reshape — it uses the raw labels
straight from the val_loader. As a result, passing a `val_loader` with
`num_classes == 2` raises a shape mismatch in `BCEWithLogitsLoss` as soon
as validation runs.

### Steps to reproduce
Train any binary-classification model with `ClassificationTrainer`,
passing a `val_loader` (e.g. via `MLPClassifier(early_stopping=True)`
in #240).

### Where I found it
While building `MLPClassifier`/`MLPRegressor` in #240 — see the `xfail`
test `TestMLPClassifierBinary.test_early_stopping_runs` in
`tests/unit/nn/test_mlp.py` for a minimal repro and full explanation.

### Suggested fix
Mirror the training branch's reshape in the validation branch of
`ClassificationTrainer.fit()`.

Related: #230

### PR DESCRIPTION
**Pull request by hkJerryLeung**
Fixes #250

## Summary
- reshape binary classification labels before validation loss calculation so BCEWithLogitsLoss receives the expected shape
- keep validation accuracy labels flat after preparing loss labels
- promote the MLP binary early-stopping regression test from xfail to an active test

## Tests
- `UV_FROZEN=1 uv run --frozen --python 3.12 nox -s unit` (605 passed, 1 skipped)
- `uv run --frozen --python 3.12 ruff check fenn`
- `uv run --frozen --python 3.12 ruff format --check fenn tests/unit/nn/test_classification_trainer.py tests/unit/nn/test_mlp.py`

### CODE REVIEW

**Comment by ApusBerliozi** [comment]
Hey @blkdmr !

Could you please take a look at these changes?

**Review by blkdmr — APPROVED** [review]
Great job! Thank you very much for your support!

**Comment by blkdmr** [comment]
@hkJerryLeung if you like the project, please star it to help us grow! Also, feel free to join our Discord server!
